src/util.js

```diff
@@ -200,7 +200,9 @@ function isPreviousLineEmpty(text, node) {
   let idx = locStart(node) - 1;
   idx = skipSpaces(text, idx, { backwards: true });
   idx = skipNewline(text, idx, { backwards: true });
-  return hasNewline(text, idx, { backwards: true });
```

Comment:
The issue is that `hasNewLine` does `opts.backwards ? index - 1 : index` so it skips one character. We need to skip that character when you are the first caller (see `-1` 3 lines above) but we don't want to do it for intermediate processing. Since `hasNewLine` is also exported it needs to do that.

This isn't a really great situation but I don't expect this part of the codebase to change that much in the future so it's good for now.

This PR inlines the implementation of hasNewLine.